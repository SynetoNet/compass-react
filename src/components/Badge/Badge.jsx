import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import BootstrapBadge from "react-bootstrap/Badge";

import "./Badge.scss";

const Badge = ({ color, variant, width, block, ...props }) => {
  const classes = classNames({
    outline: variant === "outline",
    strong: variant === "strong",
    ["width-" + width]: width,
    block
  });

  return <BootstrapBadge variant={color} className={classes} {...props} pill />;
};

Badge.propTypes = {
  color: PropTypes.oneOf(["success", "danger", "warning", "info", "secondary"]),
  variant: PropTypes.oneOf(["default", "strong", "outline"]),
  width: PropTypes.oneOf([32, 64, 96, 128, "auto"]),
  /** Makes badge stretch to the size of it's parent */
  block: PropTypes.bool
};

Badge.defaultProps = {
  color: "secondary",
  variant: "default",
  width: "auto",
  block: false
};

export default Badge;
