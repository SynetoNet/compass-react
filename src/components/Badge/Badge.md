### **Description**

---

#### **Usage**

Used to communicates various objects states or exceptional situations.

#### **Anatomy**

A badge is composed of:<br/>
Label (regular text)<br/>
Background (color fill)<br/>

### **Color variations**

---

```jsx
<h5>Strong Background</h5>
<Badge color="success">Green</Badge>
<Badge color="danger">Red</Badge>
<Badge color="warning">Yellow</Badge>
<Badge color="primary">Blue</Badge>
<Badge color="secondary">Neutral</Badge>
<br />
<br />
<h5>Light Background</h5>
<Badge color="success" lightBackground>Green</Badge>
<Badge color="danger" lightBackground>Red</Badge>
<Badge color="warning" lightBackground>Yellow</Badge>
<Badge color="primary" lightBackground>Blue</Badge>
<Badge color="secondary" lightBackground>Neutral</Badge>
<br />
<br />
<h5>Light Border</h5>
<Badge color="success" lightBorder>Green</Badge>
<Badge color="danger" lightBorder>Red</Badge>
<Badge color="warning" lightBorder>Yellow</Badge>
<Badge color="primary" lightBorder>Blue</Badge>
<Badge color="secondary" lightBorder>Neutral</Badge>
```

### **Sizes**

---

```jsx
<br />
<br />
<h5>Modular Size Badges</h5>
<Badge color="success" modularWidth widthSize={32}>32px</Badge>
<br />
<Badge color="success" modularWidth widthSize={64}>64px</Badge>
<br />
<Badge color="success" modularWidth widthSize={96}>96px</Badge>
<br />
<Badge color="success" modularWidth widthSize={128}>128px</Badge>
<br />
<br />
<h5>Elastic Size Badges</h5>
<Badge color="success" elasticWidth>Short</Badge>
<br />
<Badge color="warning" elasticWidth>Medium Size</Badge>
<br />
<Badge color="danger" elasticWidth>Longer ass Badge</Badge>
<br />
<br />
```
