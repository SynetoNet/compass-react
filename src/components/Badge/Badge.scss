@import "../../base/colors";
@import "../../base/variables";

.badge {
  height: 20px;
  width: auto;
  display: inline-block;
  padding: 0 12px;
  font-size: 12px;
  font-weight: normal;
  line-height: 20px;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;

  &.outline {
    // it contains borders, so content is only 22px height
    line-height: 18px;
  }
  &.block {
    display: block;
  }
}

.badge-success {
  background-color: $semantic-confirm-light-85;
  color: $semantic-confirm-dark-30;
  &.strong {
    background-color: $semantic-confirm;
    color: $neutral-0;
  }
  &.outline {
    background-color: $neutral-0;
    border: 1px solid $semantic-confirm-light-60;
  }
}
.badge-danger {
  background-color: $semantic-danger-light-85;
  color: $semantic-danger-dark-30;
  &.strong {
    background-color: $semantic-danger;
    color: $neutral-0;
  }
  &.outline {
    background-color: $neutral-0;
    border: 1px solid $semantic-danger-light-60;
  }
}
.badge-info {
  background-color: $semantic-info-light-85;
  color: $semantic-info-dark-30;
  &.strong {
    background-color: $semantic-info;
    color: $neutral-0;
  }
  &.outline {
    border: 1px solid $semantic-info-light-60;
    background-color: $neutral-0;
  }
}
.badge-warning {
  background-color: $semantic-warn-light-85;
  color: $semantic-warn-dark-30;
  &.strong {
    background-color: $semantic-warn;
    color: $neutral-0;
  }
  &.outline {
    background-color: $neutral-0;
    border: 1px solid $semantic-warn-light-60;
  }
}
.badge-secondary {
  background-color: $neutral-12;
  color: $neutral-90;
  &.strong {
    background-color: $neutral-60;
    color: $neutral-0;
  }
  &.outline {
    background-color: $neutral-0;
    border: 1px solid $neutral-12;
    color: $neutral-40;
  }
}

.width-32 {
  width: $spacer * 8;
  padding: 0 0;
}
.width-64 {
  width: $spacer * 16;
  padding: 0 0;
}
.width-96 {
  width: $spacer * 24;
  padding: 0 0;
}
.width-128 {
  width: $spacer * 32;
  padding: 0 0;
}
.width-160 {
  width: $spacer * 40;
  padding: 0 0;
}
.width-auto {
  width: auto;
}
