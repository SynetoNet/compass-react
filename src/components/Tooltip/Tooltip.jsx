import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import OverlayTrigger from "react-bootstrap/OverlayTrigger";
import BSTooltip from "react-bootstrap/Tooltip";
import "./Tooltip.scss";

class Tooltip extends React.Component {
  render() {
    const { text, placement, trigger, ...props } = this.props;
    return (
      <OverlayTrigger
        placement={placement}
        show={"false"}
        trigger={trigger}
        overlay={<BSTooltip>{text}</BSTooltip>}
      >
        {this.props.children}
      </OverlayTrigger>
    );
  }
}

Tooltip.propTypes = {
  text: PropTypes.string.isRequired,
  placement: PropTypes.oneOf(["top", "bottom", "left", "right", "auto"]),
  trigger: PropTypes.oneOf(["focus", "click", "hover"])
};

Tooltip.defaultProps = {
  placement: "auto",
  trigger: "hover"
};

export default Tooltip;
