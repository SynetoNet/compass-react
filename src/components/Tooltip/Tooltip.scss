@import "../../base/colors";
.tooltip-light {
  .tooltip-inner {
    color: $neutral-100;
    background-color: $neutral-0;
    border: 1px solid $neutral-100;
  }
}
