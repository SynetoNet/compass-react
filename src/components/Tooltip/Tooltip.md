#### **Badge Example on hover**

```jsx
import Badge from "../Badge/Badge";

<Tooltip placement="top" text="Some additional info" trigger="hover">
  <Badge variant="strong">Over over me!</Badge>
</Tooltip>;
```

#### **Button Example on click**

```jsx
import Button from "../Button/Button";

<Tooltip placement="right" text="Some additional info" trigger="click">
  <Button variant="primary">Click me!</Button>
</Tooltip>;
```

#### **Button Example on focus**

```jsx
import Button from "../Button/Button";
<Tooltip placement="left" text="Some additional info" trigger="focus">
  <Button variant="primary">Click me!</Button>
</Tooltip>;
```
