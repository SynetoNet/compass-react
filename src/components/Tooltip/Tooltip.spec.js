import React from "react";
import Tooltip from "./Tooltip";
import Badge from "../Badge/Badge.jsx";
import renderer from "react-test-renderer";

describe("Default Tooltip", () => {
  const defaultExample = (
    <Tooltip text="Some additional info">
      <Badge role="strong">Over over me!</Badge>
    </Tooltip>
  );

  test("tooltip snapshop", () => {
    const component = renderer.create(defaultExample);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
