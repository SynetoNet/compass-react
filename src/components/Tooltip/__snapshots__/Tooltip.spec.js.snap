// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Tooltip popover snapshop 1`] = `
<OverlayTrigger
  defaultShow={false}
  overlay={
    <ForwardRef
      className="tooltip-dark"
      placement="right"
    >
      Some additional info
    </ForwardRef>
  }
  placement="auto"
  trigger="hover"
>
  <Badge
    appearance="secondary"
    block={false}
    role="strong"
    varroleiant="default"
    width="auto"
  >
    Over over me!
  </Badge>
</OverlayTrigger>
`;
